### ISSUE TITLE
When using --map_tabix option, the output is empty despite the elements having a good score

### ISSUE THREAD

**Issue by Martingales** [adamewing/tebreak#11]
Hi Adam,

I'd like to make use of the latest mappability track updates for mm10 from https://bismap.hoffmanlab.org/ where I use the multi-read 100mer track. However, when I call _tebreak_ on a well-known sample, the resulting pickle is empty.

This is my _tebreak_ call:

```
$PYTHONVERSION $TB/tebreak/tebreak.py \
            -i $OUTPUT.1.candidates.txt \
            -b $BAM \
            -r $REF \
            -p $n \
            -m $REF_MASK \
            --map_tabix $UMAPK100 \
            --pickle $OUTPUT.1.pickle \
            --detail_out $OUTPUT.1.tb.detail \
            --tmpdir $TMPDIRTB \
            --debug > $OUTPUT.1.out 2>&1
```

When checking manually, almost all candidate elements have a larger mean mappability score than the default threshold of 0.1. Also when I call tebreak without the --map_tabix option, _tebreak_ and _resolve_ confirm more than half of the candidates (537 out of 944).

The log file reports this for all elements I checked:

> 2018-04-12 23:35:08,581 1:17253666-17253934 filtered due to min_mappability: 0 < 0

Could there be a bug in the _avgmap_ function?

**Comment by adamewing** [comment]
From the error message ("1:172...") it looks like the chromosome names in your reference genome don't have the "chr" prefix but the chromosome names in the mappability track you're using do have the "chr" prefix. So if there no chromosome matching "1" in the mappability file (because it's called "chr1") tebreak will assume the mappability is 0. Probably the easiest fix is just to remove the "chr" prefix from the mappability track.

**Comment by Martingales** [comment]
Oh wow. How embarrassing.

Thanks for pointing this out!

**Comment by Martingales** [comment]
Hi Adam,

So I have tested the mappability filter with a high minimum mappability of 0.95 on two samples and when I compare the results to the results of the standard run without mappability filtering, the results after resolve.py are exactly the same. The difference seems to be that tebreak filters more elements instead of the initial filters when starting resolve.py. Is that due to a general weakness in some elements?

Sample 1 no filtering:
886 candidates
507 elements after tebreak
440 elements after resolve

Sample 1 filtering with 0.95:
886 candidates
440 elements after tebreak
440 elements after resolve

Also, the logfile still shows  "0 < 0" when filtering out an element. Is that on purpose?

> 2018-04-19 16:11:28,170 1:76918721-76918991 filtered due to min_mappability: 0 < 0

**Comment by adamewing** [comment]
That seems like the expected behaviour unless I'm misunderstanding - filtering on mappability can only decrease the number of insertions detected and it's only an option for `tebreak.py`. It's interesting that the number of variants retained after `resolve.py` is unchanged but hard to guess at whether that's a coincidence... is this capture/targeted sequencing or WGS? What are the candidates and how do you know there's 886 of them before running tebreak?

The log entry is just due to me making a mistake in string substitution and using `%d` instead of `%f`, will fix, thanks.

I haven't been using mappability as a filter when I'm running tebreak using the `-d` and `-m` options (see current tebreak readme) as I find it can cause false negatives.

**Comment by Martingales** [comment]
The 886 are the number of candidates I grep from the output of the discocluster.py script which I then feed as input to tebreak.py (option -i).

The samples are 40x WGS. My challenge is that I have almost 700 samples. That's why I need to use the discocluster script in order to make the overall runtime somewhat feasible.

**Comment by adamewing** [comment]
Ah, ok, so the `-d/--disco_target` option is the equivalent of using the "discocluster" script.

**Comment by adamewing** [comment]
... and I'd highly recommend it instead of using the discocluster script.

**Comment by Martingales** [comment]
Thanks for the recommendation!

I changed my code to use only the -d option but I realised that when looping over multiple samples, the disc.debug.txt and sample.tebreak.chunklist.out are written one level higher than all other output files, where each sample process overwrites the disc.debug.txt file. Is that file important?
Have I missed a command line option where I specify the output files for the -d option?
